crates/alpen-ee/common/src/types/chunk.rs

```diff
@@ -0,0 +1,83 @@
+use std::iter;
+
+use strata_acct_types::Hash;
+
+/// Lifecycle states for chunk
+#[derive(Debug)]
+pub enum ChunkStatus {
+    /// Proving has not started yet.
+    Unproven,
+    /// Proving started. Pending proof generation.
+    PendingProof(String),
+    /// Valid proof ready.
+    /// TODO: correct proof type
+    ProofReady(Vec<u8>),
+}
+
+/// Unique identifier for a chunk
+#[derive(Debug)]
+pub struct ChunkId {
+    pub prev_block: Hash,
+    pub last_block: Hash,
+}
+
+/// Represents a sequence of blocks that are processed together as a unit during proving.
+#[derive(Debug)]
+pub struct Chunk {
+    /// Sequential chunk index
+    idx: u64,
+    /// last block of (idx - 1)th chunk that this chunk extends
+    prev_block: Hash,
+    /// last block of this chunk. A chunk cannot be empty.
+    last_block: Hash,
+    /// rest of the blocks in the chunk.
+    blocks: Vec<Hash>,
```

Comment:
Do we need to grow this dynamically?
Otherwise, we could do

```suggestion
    /// rest of the blocks in the chunk.
    blocks: &[Hash],
```